google-cloud-spanner/src/main/java/com/google/cloud/spanner/Mutation.java

```diff
@@ -464,7 +494,32 @@ static void toProto(Iterable<Mutation> mutations, List<com.google.spanner.v1.Mut
     }
     // Flush last item.
     if (proto != null) {
+      com.google.spanner.v1.Mutation builtMutation = proto.build();
       out.add(proto.build());
+      if (checkIfInsertMutationWithLargeValue(builtMutation, largeInsertMutation)) {
+        largeInsertMutation = builtMutation;
+      }
+      if (!builtMutation.hasInsert()) {
+        allMutationsExcludingInsert.add(builtMutation);
+      }
     }
+
+    // Select a random mutation based on the heuristic.
+    if (allMutationsExcludingInsert.size() > 0) {
+      int randomIndex = ThreadLocalRandom.current().nextInt(allMutationsExcludingInsert.size());
+      return allMutationsExcludingInsert.get(randomIndex);
```

Comment:
nit: combine this into one line (meaning; skip the declaration of `int randomIndex`)

Reply:
Done